audmetric/core/api.py

```diff
@@ -114,20 +120,37 @@ def concordance_cc(
     if len(prediction) < 2:
         return np.NaN
 
-    r = pearson_cc(prediction, truth)
-    x_mean = prediction.mean()
-    y_mean = truth.mean()
-    x_std = prediction.std()
-    y_std = truth.std()
-    denominator = (
-        x_std * x_std
-        + y_std * y_std
-        + (x_mean - y_mean) * (x_mean - y_mean)
-    )
+    # Handle mask NaN cases separetly
+    # to be as fast as possible
+    consider_nan = False
+    if ignore_nan:
+        mask = ~(np.isnan(truth) + np.isnan(prediction))
```

Comment:
Why can't we simply do:

```python
if ignore_nan:
    mask = ~(np.isnan(truth) + np.isnan(prediction))
    truth = truth[mask]
    prediction = prediction[mask]
```

and then execute the same code for both conditions?

Reply:
I mean, I understand that the new implementation is faster than before, but does applying the mask twice really leads to an additional slow-down?

Reply:
It changes for me from 0.23 s to 0.34 s.

Reply:
On 10 million samples right? I mean if you have that many samples in your training a speed up of 0.1s is nothing :)

Reply:
@dkounadis can you please elaborate why speed was important for you when estimating CCC?

Reply:
What I mean is that everything else will take so much longer that you will most likely not notice it.

Reply:
Ok, seems we are bit stuck here. So I propose the following: we continue with the proposed solution. But before we release a new version we create another PR where we add `ignore_nat` at least to `pearson_cc()`. There we can might keep it simple and simply add:

```
if ignore_nan:
    mask = ~(np.isnan(truth) + np.isnan(prediction))
    truth = truth[mask]
    prediction = prediction[mask]
```

Reply:
As we did a release in the mean time it would be also fine with me to add support for `ignore_nan` to all functions where it makes sense.

Regarding the speed I would still be interested to hear from @dkounadis how he is using it.

Reply:
I discussed with @dkounadis offline, and he is fine with my changes here.